sktime/annotation/stray.py

```diff
@@ -0,0 +1,283 @@
+        Sample size to calculate an emperical threshold.
+    outlier_tail : str {"min", "max"}, optional (default="max")
+        Direction of the outlier tail.
+
+    Attributes
+    ----------
+    score_ : pd.Series
+        Outlier score of each data point in X.
+    y_ : pd.Series
+        Outlier boolean flag for each data point in X.
+
+    References
+    ----------
+    [1] Talagala, Priyanga Dilini, Rob J. Hyndman, and Kate Smith-Miles.
+    "Anomaly detection in high-dimensional data." Journal of Computational
+    and Graphical Statistics 30.2 (2021): 360-374.
+    [2] Wilkinson, Leland. "Visualizing big data outliers through
+    distributed aggregation." IEEE transactions on visualization and
+    computer graphics 24.1 (2017): 256-266.
+
+    Examples
+    --------
+    >>> from sktime.annotation.stray import STRAY
+    >>> from numpy import asarray
+    >>> X = asarray([3.7,3.2,3.4,3.6,-5.1,-5.2,-4.9])
+    >>> model = STRAY(k=3)
+    >>> y = model.fit_transform(X)
+    >>> y
+    array([0, 0, 0, 0, 1, 1, 1])
+    """
+
+    _tags = {
+        "handles-missing-data": True,
+        "X_inner_mtype": "np.ndarray",
+        "fit_is_empty": False,
+        "skip-inverse-transform": True,
+    }
+
+    def __init__(
+        self,
+        alpha=0.01,
+        k=10,
+        knn_algorithm="brute",
+        normalize=unitize,
+        p=0.5,
+        size_threshold=50,
+        outlier_tail="max",
+    ):
+        self.alpha = alpha
+        self.k = k
+        self.knn_algorithm = knn_algorithm
+        self.normalize = normalize
+        self.p = p
+        self.size_threshold = size_threshold
+        self.outlier_tail = outlier_tail
+        super(STRAY, self).__init__()
+
+    def _find_threshold(self, outlier_score, n):
+        """Find Outlier Threshold.
+
+        Parameters
+        ----------
+        outlier_score : np.ArrayLike
+            The outlier scores determined by k nearast neighbours distance
+        n : int
+            The number of rows remaining in X when NA's are removed.
+
+        Returns
+        -------
+        array of indices of the observations determined to be outliers.
+        """
+        if self.outlier_tail == "min":
+            outlier_score = -outlier_score
+
+        order = np.argsort(outlier_score)
+        gaps = np.append(0, np.diff(outlier_score[order]))
+        n4 = int(max(min(self.size_threshold, np.floor(n / 4)), 2))
+
+        J = np.array([i for i in range(2, n4 + 1)])
+        start = int(max(np.floor(n * (1 - self.p)), 1))
+
+        ghat = [
+            0.0 if i < start else sum((J / (n4 - 1)) * gaps[i - J + 1])
+            for i in range(n)
+        ]
+
+        log_alpha = np.log(1 / self.alpha)
+        bound = np.Inf
+
+        for i in range(start, n):
+            if gaps[i] > log_alpha * ghat[i]:
+                bound = outlier_score[order][i - 1]
+                break
+
+        return np.where(outlier_score > bound)[0]
+
+    def _use_KNN(self, X, n):
```

Comment:
Consider changing the name to something a bit more descriptive, e.g. `find_nearest_neighbors`.
```suggestion
    def _use_KNN(self, X: npt.ArrayLike, n: int) -> Dict:
```